apps/antalmanac/src/routes/AuthPage.tsx

```diff
@@ -0,0 +1,97 @@
+import { useEffect, useCallback } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import { isEmptySchedule, mergeShortCourseSchedules } from '$actions/AppStoreActions';
+import { LoadingScreen } from '$components/LoadingScreen';
+import trpc from '$lib/api/trpc';
+import {
+    getLocalStorageDataCache,
+    getLocalStorageFromLoading,
+    setLocalStorageImportedUser,
+    getLocalStorageUserId,
+    removeLocalStorageUserId,
+    removeLocalStorageImportedUser,
+    removeLocalStorageDataCache,
+    removeLocalStorageFromLoading,
+    setLocalStorageSessionId,
+} from '$lib/localStorage';
+import AppStore from '$stores/AppStore';
+import { useSessionStore } from '$stores/SessionStore';
+
+export function AuthPage() {
+    const { session } = useSessionStore();
+    const [searchParams] = useSearchParams();
+    // const navigate = useNavigate();
+
+    const handleSearchParamsChange = useCallback(async () => {
+        try {
+            const code = searchParams.get('code');
+            if (!code) {
+                window.location.href = '/';
+                return;
+            }
+
+            const { sessionToken, userId, providerId } = await trpc.userData.handleGoogleCallback.mutate({
+                code: code,
+                token: session ?? '',
+            });
+
+            const fromLoading = getLocalStorageFromLoading() ?? '';
+            const savedUserId = getLocalStorageUserId() ?? '';
+            const savedData = getLocalStorageDataCache() ?? '';
+            removeLocalStorageUserId();
+
+            if (sessionToken && providerId) {
```

Comment:
These cascading `if`s are nighnmarish. You should break it up into helpfully-named callbacks to make it both easier to parse as well as to give names to the actions you are performing logically.